### PR TITLE
Properly serialize error responses in unity web request adapter

### PR DESCRIPTION
**Pull request by lugehorsam**
Companion PR to https://github.com/heroiclabs/nakama-dotnet/pull/37 that applies its fix to the `UnityWebRequestAdapter`.

Also included a new `nakama-dotnet` DLL with the extension method included.

### CODE REVIEW

File: Packages/Nakama/Runtime/UnityWebRequestAdapter.cs [hunk 1]
```diff
@@ -121,9 +121,20 @@ private UnityWebRequestAdapter()
...
                 var decoded = www.downloadHandler.text.FromJson<Dictionary<string, object>>();
-                errback(decoded == null
-                    ? new ApiResponseException(www.downloadHandler.text)
-                    : new ApiResponseException(www.responseCode, decoded["error"].ToString(), (int) decoded["code"]));
+
+                ApiResponseException e;
+
+                if (decoded == null)
+                {
+                    e = new ApiResponseException(www.downloadHandler.text);
```
Comment (novabyte) [inline Packages/Nakama/Runtime/UnityWebRequestAdapter.cs:129]: You could move this initialization to line 125. Then check if decoded not equal to `null` and replace the object.

Comment (lugehorsam) [reply Packages/Nakama/Runtime/UnityWebRequestAdapter.cs:129]: @novabyte I don't mind doing that -- I was wondering if that is generally considered more readable? That approach would be less lines of code. On the other hand, the advantage of current approach is it's only getting assigned to once which creates one less object. (Obviously we are talking fine grains of sand here, but I was just curious.)

Comment (novabyte) [reply Packages/Nakama/Runtime/UnityWebRequestAdapter.cs:129]: I think in general we always optimize to minimize allocations but in this case its an exceptional code path (low active execution) and will always allocate one object but many eliminate one overwrite with the allocation so its all small impact. :)

**Comment by lugehorsam** [comment]
@novabyte ready for another look

**Comment by novabyte** [comment]
Thanks @lugehorsam 👍

**Pull request merged**